src/debugSession/BrightScriptDebugSession.ts

```diff
@@ -670,21 +670,135 @@ export class BrightScriptDebugSession extends BaseDebugSession {
-            if (i < lines.length - 1) {
-                line += '\n';
+
+        this.pendingSendLogPromise = this.pendingSendLogPromise.then(async () => {
+            logOutput = await this.convertBacktracePaths(logOutput);
+
+            const lines = logOutput.split(/\r?\n/g);
+            for (let i = 0; i < lines.length; i++) {
+                let line = lines[i];
+                if (i < lines.length - 1) {
+                    line += '\n';
+                }
+
+                if (this.launchConfiguration.enableConsolePkgToSourcePathConversion) {
+                    let potentialPaths = this.getPotentialPkgPaths(line);
+                    for (let potentialPath of potentialPaths) {
+                        let originalLocation = await this.projectManager.getSourceLocation(potentialPath.path, potentialPath.lineNumber, potentialPath.columnNumber);
+                        if (originalLocation) {
+                            let replacement: string;
+                            if (originalLocation.filePath.startsWith('/')) {
+                                replacement = `file:/${originalLocation.filePath}:${originalLocation.lineNumber}`;
+                            } else {
+                                replacement = `file://${originalLocation.filePath}:${originalLocation.lineNumber}`;
+                            }
+
+                            if (potentialPath.columnNumber !== undefined) {
+                                replacement += `:${originalLocation.columnIndex + 1}`;
+                            }
+
+                            line = line.replaceAll(potentialPath.fullMatch, replacement);
+                        }
+                    }
+                }
+                this.sendEvent(new OutputEvent(line, 'stdout'));
+                this.sendEvent(new LogOutputEvent(line));
+            }
+        });
+        return this.pendingSendLogPromise;
+    }
+
+    // https://regex101.com/r/ixpQiq/1
+    private deviceFilePathRegex = /((?:\.\.\.|[A-Za-z_0-9]*pkg\:\/)[A-Za-z_0-9 \/\.]+\.[A-Za-z_0-9 \/]+)(?:(?:\:)(\d+)(?:\:(\d+))?|\((\d+)(?:\:(\d+))?\))/ig;
+
+    /**
+     * Extracts potential package paths from a given line of text.
+     *
+     * This method uses a regular expression to find matches in the provided line
+     * and returns an array of objects containing details about each match.
+     *
+     * @param input - The line of text to search for potential package paths.
+     * @returns An array of objects, each containing:
+     *   - `fullMatch`: The full matched string.
+     *   - `path`: The extracted path from the match.
+     *   - `lineNumber`: The line number extracted from the match.
+     *   - `columnNumber`: The column number extracted from the match, or `undefined` if not found.
+     */
+    private getPotentialPkgPaths(input: string): Array<{ fullMatch: string; path: string; lineNumber: number; columnNumber: number }> {
+        let matches = input.matchAll(this.deviceFilePathRegex);
```

Comment:
let's just inline the regex instead of having it be a prop. v8 optimizes this so it's just as fast as caching it.